### PR TITLE
Created newer version (2018-06-17-preview) of workbook resource type

### PR DESCRIPTION
**Pull request by ericc1103**
Added sourceId as an optional parameter.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jianghaolu** [comment]
@ericc1103 Sorry for the delay - do you have plans to add the operation for `/providers/microsoft.insights/operations`?

**Comment by jianghaolu** [comment]
Hi @ericc1103 Sorry for the late reply.

You can find more details about the required operation here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3023-operationsapiimplementation.

**Comment by ericc1103** [comment]
@jianghaolu I have added `Microsoft/Insights/Operations`. Let me know if I missed anything.

**Comment by jianghaolu** [comment]
Please add the new spec files to REAME.md, thanks!

**Comment by ericc1103** [comment]
@jianghaolu please review this and let's finalize it. Thanks!

**Comment by ravbhatnagar** [comment]
This was reviewed over skype.

**Comment by ericc1103** [comment]
@jianghaolu Ping again...

**Comment by anuchandy** [comment]
@ericc1103 can you ensure these fixes https://github.com/Azure/azure-rest-api-specs/pull/3324 are ported to this new api-version swagger as well?

Also there seems conflict in the readme file.

**Comment by ericc1103** [comment]
@anuchandy Applied these fixes #3324 and resolved the merge conflicts.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-06-17-preview/workbooks_API.json [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+            },
+            "description": "Resource tags"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
```
Comment (anuchandy) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-06-17-preview/workbooks_API.json]: Can user change the location of workbook resource? if not please remove this property.

**Comment by anuchandy** [comment]
@ericc1103 thank you, i've a question on location update,  please see above comment. Once you address it we are good.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-06-17-preview/workbooks_API.json (hunk 1 above)
Comment (ericc1103) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-06-17-preview/workbooks_API.json]: @anuchandy  Done!

**Comment by anuchandy** [comment]
@ericc1103 actually we need to remove the location property from the update model if location cannot be changed.

**Comment by anuchandy** [comment]
@ericc1103 looks good. One more question - is this API version deployed to production at least in one region?

**Review by anuchandy — APPROVED** [review]

**Pull request merged**